src/com/mediaengine/rpc/rpc_cdm_mediaengine_handler.cc

```diff
@@ -50,6 +71,170 @@ RpcCdmMediaengineHandler::RpcCdmMediaengineHandler() {
 
 }
 
+
+#ifdef OCDM_SDP_PROTOTYPE
+
+static int connectSocket(void)
+{
+  int status = 0;
+  int socket_fd = -1;
+  struct sockaddr_un socketAddress;
+  struct timeval timeout_tv;
+
+  socket_fd = socket(AF_UNIX, SOCK_STREAM, 0);
+  if(socket_fd < 0) {
+    CDM_DLOG() << "connectSocket: Failure to create socket";
+    status = -1;
+    goto handle_error;
+  }
+
+  /* Use abstract socket (First byte is \0). */
+  memset(&socketAddress, 0, sizeof(socketAddress));
+  socketAddress.sun_family = AF_UNIX;
+  strcpy(&socketAddress.sun_path[1], "opencdm_secure_fd_communication");
+
+  if(connect(socket_fd, (struct sockaddr *)&socketAddress, sizeof(socketAddress)) < 0) {
+    CDM_DLOG() << "connectSocket: Failure to connect socket";
+    status = -1;
+    goto handle_error;
+  }
+
+  /* Configure a timeout for sendmsg(). This is to avoid an infinite wait. */
+  timeout_tv.tv_sec = SOCKET_SEND_TIMEOUT;
+  timeout_tv.tv_usec = 0;
+  status = setsockopt(socket_fd, SOL_SOCKET, SO_SNDTIMEO, (const void *)&timeout_tv, sizeof(timeout_tv));
+  if(status < 0) {
+    CDM_DLOG() << "connectSocket: Cannot configure sendmsg timeout";
+    goto handle_error;
+  }
+
+handle_error:
+  if(status < 0) {
+    if(socket_fd >= 0) {
+      close(socket_fd);
+      socket_fd = -1;
+    }
+  }
+  return socket_fd;
+}
+
+
+// The caller is responsible to close the returned file descriptor.
+static int allocateIon(size_t size)
+{
+  struct ion_allocation_data alloc_data;
+  struct ion_handle_data hdl_data;
+  struct ion_fd_data fd_data;
+  int ion;
+  int fd = -1;
+
+  ion = open("/dev/ion", O_RDWR);
```

Comment:
We most likely need some way of detecting which ION API the kernel is exposing and either supporting both (best case) for the buffer allocation. Or alternatively erroring nicely if the kernel is newer than what we support.  I think xtest --sdp-basic unit test already has some method for trying to do this detection (although IIRC I think it was actually broken).

Reply:
I would like to handle this comment separately. I have created a separate JIRA ticket (MMWG-48) for that purpose.